src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,6 +181,87 @@ private string GetRootPathWithGuid_NoLock()
             return new MetadataAsSourceFile(fileInfo.TemporaryFilePath, navigateLocation, documentName, documentTooltip);
         }
 
+        private async Task<Document> DecompileSymbolAsync(Document temporaryDocument, ISymbol symbol, CancellationToken cancellationToken)
+        {
+            // Get the name of the type the symbol is in:
+            var containingOrThis = symbol.GetContainingTypeOrThis();
+            var total = GetFullReflectionName(containingOrThis);
+
+            var model = await temporaryDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            // TODO : retrieve path to actual assembly instead of reference assembly:
+            var reference = model.Compilation.GetMetadataReference(symbol.ContainingAssembly);
+            // Load the assembly.
+            var ad = AssemblyDefinition.ReadAssembly(reference.Display, new ReaderParameters() { AssemblyResolver = new RoslynAssemblyResolver(model.Compilation) });
+
+            // Initialize a decompiler with default settings.
+            CSharpDecompiler decompiler = new CSharpDecompiler(ad.MainModule, new DecompilerSettings());
+            var fullTypeName = new FullTypeName(total);
+
+            // Try to decompile
+            // If it fails, display stack trace and assembly file name.
+            string text;
+            try
+            {
+                text = decompiler.DecompileTypeAsString(fullTypeName);
+            }
+            catch (Exception ex)
+            {
+                text = ad.MainModule.FileName + "\n" + ex.ToString();
+            }
+            return temporaryDocument.WithText(SourceText.From(text));
+        }
+
+        private class RoslynAssemblyResolver : IAssemblyResolver
+        {
+            private readonly Compilation parentCompilation;
+
+            public RoslynAssemblyResolver(Compilation parentCompilation)
+            {
+                this.parentCompilation = parentCompilation;
+            }
+
+            public AssemblyDefinition Resolve(AssemblyNameReference name)
+            {
+                return Resolve(name, new ReaderParameters());
+            }
+
+            public AssemblyDefinition Resolve(AssemblyNameReference name, ReaderParameters parameters)
+            {
+                foreach (var assembly in parentCompilation.GetReferencedAssemblySymbols())
+                {
+                    if (assembly.Identity.Name == name.Name
+                        && assembly.Identity.Version == name.Version
+                        && assembly.Identity.PublicKeyToken.SequenceEqual(name.PublicKeyToken ?? new byte[0]))
```

Comment:
Array.Empty<byte>()?

Reply:
:arrow_right: Fixed in 3a53d556e819164f8bf04409347f77b89e25f8d7